trpcultivate_phenoshare/src/Plugin/TripalImporter/TripalCultivatePhenoShareImporter.php

```diff
@@ -743,6 +914,679 @@ public function formValidate($form, &$form_state) {
+   *   input file. This unordered list will include a table that lists the row
+   *   and column combinations with empty cells. It has the following headers:
+   *   - 'Line Number'
+   *   - 'Column(s) with empty value'
+   *
+   * @throws \Exception
+   *   - If the validation_result parameter was not formatted properly.
+   *   - If the case string returned by the validator implied validation passed.
+   *   - If the case string returned by the validator is not recognized.
+   */
+  public function processEmptyCellFailures(array $failures) {
+    // Define our table header.
+    $table_header = ['Line Number', 'Column(s) with empty value'];
+    $table['rows'] = [];
+
+    foreach ($failures as $line_no => $validation_result) {
+      // Check the format of the validation_result parameter.
+      $this->checkValidationStatusArray($validation_result, 'EmptyCell', $line_no);
+
+      if ($validation_result['case'] == 'Empty value found in required column(s)') {
+        $table['message'] = 'The following line number and column header combinations were empty, but a value is required.';
+        // Convert indices in failedItems to column headers.
+        $failed_indices = $validation_result['failedItems']['empty_indices'];
+        // For each index with an empty value, grab the column name from our
+        // $headers property and add to an array of header names.
+        $empty_headers = [];
+        foreach ($failed_indices as $index) {
+          array_push($empty_headers, $this->headers[$index]['name']);
+        }
+        // Implode the empty headers array into a string and then add it as a
+        // row to our table.
+        $columns_string = implode(", ", $empty_headers);
+        array_push($table['rows'], [
+          $line_no,
+          $columns_string,
+        ]);
+      }
+      elseif ($validation_result['case'] == 'No empty values found in required column(s)') {
+        throw new \Exception("The case string returned by the EmptyCell validator at line #$line_no implies validation passed, but valid is set to FALSE.");
+      }
+      else {
+        throw new \Exception("The case string returned by the EmptyCell validator at line #$line_no is not recognized as a potential case.");
+      }
+    }
+
+    // Build the render array for our table.
+    $render_array = [
+      '#theme' => 'item_list',
+      '#type' => 'ul',
+      '#attributes' => [
+        'class' => [
+          'tcp-empty-cell-failures',
+        ],
+      ],
+      '#items' => [
+        [
+          [
+            '#prefix' => '<div class="case-message">',
+            '#markup' => $table['message'],
+            '#suffix' => '</div>',
+          ],
+          [
+            '#type' => 'table',
+            '#header' => $table_header,
+            '#attributes' => [],
+            '#rows' => $table['rows'],
+          ],
+        ],
+      ],
+    ];
+
+    return $render_array;
+  }
+
+  /**
+   * Use documentation in the traits importer.
```

Comment:
It looks like this docblock still needs updating

Reply:
Added in commit [4b785ea](https://github.com/TripalCultivate/TripalCultivate-Phenotypes/pull/150/commits/4b785eaefb5ca66669a21a13bb4a51aaaa3de283)